toolchain/check/testdata/operators/overloaded/index_with_prelude.carbon

```diff
@@ -34,8 +34,10 @@ let x: ElementType = c[s];
 
 library "[[@TEST_NAME]]";
 
-impl (i32, i32) as Core.IndexWith(i32, i32) {
-  fn At[self: Self](subscript: i32) -> i32 {
+// TODO: Replace `IntLiteral` with `i32`. Right now that's not handled by impl
+// lookup.
```

Comment:
What is the failure / what are you expecting here?

Are you expecting a Core.IntLiteral to convert to i32 to find a specialization? I think are intentional in not doing any conversions. Maybe this would need to be written as a blanket impl with `T: ImplicitAs(i32)` instead?